external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

```diff
@@ -127,12 +132,19 @@ public void open(Map conf, TopologyContext context, SpoutOutputCollector collect
     // =========== Consumer Rebalance Listener - On the same thread as the caller ===========
 
     private class KafkaSpoutConsumerRebalanceListener implements ConsumerRebalanceListener {
+        private ConsumerRebalanceHandlerTask task;
+
+        public KafkaSpoutConsumerRebalanceListener(ConsumerRebalanceHandlerTask task) {
+            this.task = task;
+        }
+
         @Override
         public void onPartitionsRevoked(Collection<TopicPartition> partitions) {
             LOG.info("Partitions revoked. [consumer-group={}, consumer={}, topic-partitions={}]",
                     kafkaSpoutConfig.getConsumerGroupId(), kafkaConsumer, partitions);
-            if (!consumerAutoCommitMode && initialized) {
-                initialized = false;
+
+            if (!consumerAutoCommitMode && task.isComplete()) {
```

Comment:
it's like an boolean, asynchronous task. start and stop will set the boolean field. start() -> complete=true, stop() -> complete=false